plugins/lookup/etcd3.py

```diff
@@ -0,0 +1,230 @@
+            type: list
+            elements: str
+            required: True
+        prefix:
+            description:
+            - look for key or prefix key
+            type: bool
+        host:
+            description:
+            - etcd3 listening client host
+            env:
+            - name: ETCDCTL_ENDPOINTS
+            default: '127.0.0.1'
+            type: str
+        port:
+            description:
+            - etcd3 listening client port
+            default: 2379
+            type: int
+        ca_cert:
+            description:
+            - etcd3 CA authority
+            default: None
+            env:
+            - name: ETCDCTL_CACERT
+            type: str
+        cert_cert:
+            description:
+            - etcd3 client certificate
+            default: None
+            env:
+            - name: ETCDCTL_CERT
+            type: str
+        cert_key:
+            description:
+            - etcd3 client private key
+            default: None
+            env:
+            - name: ETCDCTL_KEY
+            type: str
+        timeout:
+            description:
+            - client timeout
+            default: 60
+            env:
+            - name: ETCDCTL_DIAL_TIMEOUT
+            type: int
+        user:
+            description:
+            - auth user name
+            default: None
+            env:
+            - name: ETCDCTL_USER
+            type: str
+        password:
+            description:
+            - auth user password
+            default: None
+            env:
+            - name: ETCDCTL_PASSWORD
+            type: str
+
+    requirements:
+    - "etcd3 >= 0.10"
+'''
+
+EXAMPLES = '''
+    - name: "a value from a locally running etcd"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo/bar') }}"
+
+    - name: "values from multiple folders on a locally running etcd"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo', 'bar', 'baz') }}"
+
+    - name: "look for a key prefix"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', '/foo/bar', prefix=True) }}"
+
+    - name: "connect to etcd3 with a client certificate"
+      debug:
+        msg: "{{ lookup('community.general.etcd3', 'foo/bar', cert_cert='/etc/ssl/etcd/client.pem', cert_key='/etc/ssl/etcd/client.key') }}"
+'''
+
+RETURN = '''
+    _raw:
+        description:
+        - List of keys and associated values.
+        type: list
+        elements: complex
+'''
+
+import os
+import re
+import traceback
+
+from ansible.errors import AnsibleLookupError
+from ansible.utils.display import Display
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils._text import to_native
+
+try:
+    import etcd3
+    HAS_ETCD = True
+except ImportError:
+    HAS_ETCD = False
+
+display = Display()
+
+
+def etcd3_client(client_params):
+    etcd = None
```

Comment:
```suggestion
```
Or is there a reason this assignment is made here?

Reply:
right, it is useless